napari/_qt/_qapp_model/qactions/_view.py

```diff
@@ -103,3 +104,13 @@ def _get_current_activity_dock_status(window: Window):
         toggled=ToggleRule(get_current=_get_current_activity_dock_status),
     ),
 ]
+
+if sys.platform == 'darwin':
+    Q_VIEW_ACTIONS.append(
+        Action(
+            id=f'{CommandId.TOGGLE_FULLSCREEN}_mac',
+            title=f'{CommandId.TOGGLE_FULLSCREEN.command_title}_mac',
+            callback=Window._toggle_fullscreen,
+            keybindings=[StandardKeyBinding.FullScreen],
+        )
+    )
```

Comment:
Apparently the macOS standard keybinding for Full screen post Monterey (macOS 12) was changed to Fn (Globe) F. 

We wanted to hide our full screen menu item  but have the old keybinding (Control-Command-F) work:

> I think in an ideal world, the old keybind would keep working (and be settable as a Viewer keybind), but there would be no custom menu item, just the native one?

See: https://github.com/napari/napari/issues/6240#issuecomment-1728619669

Unfortunately it seems that although the above code will register the action and the keybinding, and the action is executable manually via `app.commands.execute_command`, the keybinding is not 'active' (will not work) unless the action is added to a menu item.

Obviously we would like this to work, especially for our command palette project. Depending on why the action is not 'active' and how difficult the fix is, this may have to wait until command palette, where we would want to be able to register actions but not have to add them to a menu.

Reply:
It will not work as it is not registered. Shortcuts from the menu are handled by Qt. not by the keymap provider.

Reply:
Not registered where? `Q_VIEW_ACTIONS` all get registered with app model and other actions in `Q_VIEW_ACTIONS` with keybindings work? 
Can you tell me what the meaning of keymap provider is?

Reply:
As far as I check in the codebase. We currently use app_models actions shortcuts to define menu entry items. In such scenario the action is triggered by QMenu. 

All shortcuts that are not triggered by some menu, goes through `napari.utils.action_manager.action_namager` where shortcuts are defined based on `Settings.shortcuts.shortcuts`

https://github.com/napari/napari/blob/bf381d7ba02b6b561c9bf80fa8ad9aafd7ee593d/napari/_qt/qt_viewer.py#L435-L440


So adding action without menu position will not have any impact currently as far as I understand the codebase.

Reply:
Ahh this sounds like exactly what is happening. Thank you.

I guess [nap-7](https://napari.org/dev/naps/7-key-binding-dispatch.html) and https://github.com/napari/napari/pull/6204 is about moving keybindings to app-manager.
(I noted that this work is not in the current road map, do you think it should be? cc @DragaDoncila )

Also I guess we would need app-model keybindings to be implemented to be able to implement the command palette.